users/iam.tf

```diff
@@ -0,0 +1,72 @@
+data "aws_iam_policy_document" "transfer_assume_role" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["transfer.amazonaws.com"]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "s3_read_write" {
+  Statement = {
+    effect    = "Allow"
+    actions   = [
+      "s3:PutObject",
+      "s3:GetObject",
+      "s3:DeleteObject",
+      "s3:DeleteObjectVersion",
+      "s3:GetObjectVersion",
+      "s3:ListBucket",
+      "s3:GetBucketLocation"
+    ]
+    resources = [
+      var.s3_bucket_arn,
+      "${var.s3_bucket_arn}/*"
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "s3_read_only" {
+  Statement = {
+    effect    = "Allow"
+    actions   = [
+      "s3:GetObject",
+      "s3:GetObjectVersion",
+      "s3:ListBucket",
+      "s3:GetBucketLocation"
+    ]
+    resources = [
+      var.s3_bucket_arn,
+      "${var.s3_bucket_arn}/*"
+    ]
+  }
+}
+
+resource "aws_iam_role" "s3_read_write_role" {
+  count              = var.read_write_users ? 1 : 0
+  name               = "${var.service_name}_transfer-s3-read-write-role"
```

Comment:
Nitpick: mixing underscores `_` and dashes `-` here in the name. Same comment for the other blocks like this--let's keep it consistent please. Dashes.

Reply:
The reason we did this was for a namespacing kind of thing. It was intentional but if you'd rather not use it, that's fine. I'll just not that we have previously namespaced outputs so that it's clear what service they're coming from. I guess since this isn't an output it doesn't make sense.